tests/rest_test/rest_registry_test.cpp

```diff
@@ -0,0 +1,358 @@
+// Glaze Library
+// For the license information refer to glaze.hpp
+
+#include "glaze/glaze.hpp"
+#include "glaze/rest/rest.hpp"
+#include "glaze/rpc/registry.hpp"
+#include "ut/ut.hpp"
+
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+using namespace ut;
+
+struct User {
+   int id{};
+   std::string name{};
+   std::string email{};
+};
+
+struct UserIdRequest {
+   int id{};
+};
+
+struct ErrorResponse {
+   std::string error{};
+};
+
+// Define the UserService that will be exposed via REST
+struct UserService {
+   // State
+   std::unordered_map<int, User> users = {
+      {1, {1, "John Doe", "john@example.com"}},
+      {2, {2, "Jane Smith", "jane@example.com"}}
+   };
+   int next_id = 3;
+   
+   // Methods that will be exposed as REST endpoints
+   
+   // Get all users
+   std::vector<User> getAllUsers() {
+      std::vector<User> user_list;
+      for (const auto& [id, user] : users) {
+         user_list.push_back(user);
+      }
+      return user_list;
+   }
+   
+   // Get user by ID
+   User getUserById(UserIdRequest request) {
+      auto it = users.find(request.id);
+      if (it != users.end()) {
+         return it->second;
+      }
+      // In a real app, you'd want proper error handling
+      // For simplicity, we'll return an empty user
+      return User{};
+   }
+   
+   // Create a new user
+   User createUser(User user) {
+      user.id = next_id++;
+      users[user.id] = user;
+      return user;
+   }
+};
+
+template <>
+struct glz::meta<UserService> {
+   using T = UserService;
+   static constexpr auto value = object(&T::getAllUsers, &T::getUserById, &T::createUser);
+};
+
+// Helper function to read a file into a string
+std::string read_file(const std::string& path) {
+   std::ifstream file(path);
+   if (!file.is_open()) {
+      return "";
+   }
+   std::stringstream buffer;
+   buffer << file.rdbuf();
+   return buffer.str();
+}
+
+int main() {
+   glz::http_server server;
+   
+   UserService userService;
+   
+   glz::rest_registry registry;
+   
+   registry.on(userService);
+   
+   server.mount("/api", registry.endpoints);
+   
+   // Serve the frontend files
+   server.get("/", [](const glz::Request& /*req*/, glz::Response& res) {
```

Comment:
Any reason not have these (`glz::Request` and `glz::Response`) be snake case like the rest?

Reply:
I do want them snake case. Thanks for reminding me. I was afraid of a collision with `glz::repe::request`, but there are different namespaces. Will update.